tests/integration_tests/functional/test_config_space.py

```diff
@@ -0,0 +1,351 @@
+        "     s.getsockopt(socket.SOL_SOCKET,\n"                    \
+        "                 socket.SO_REUSEADDR) | 1\n"               \
+        ")\n"                                                       \
+        "s.bind(('{}', {}))\n"                                      \
+        "s.listen(1)\n"                                             \
+        "conn, addr = s.accept()\n"                                 \
+        "recv_iterations = {}\n"                                    \
+        "while recv_iterations > 0:\n"                              \
+        "    data = conn.recv({})\n"                                \
+        "    recv_iterations -= 1\n"                                \
+        "conn.close()\n"                                            \
+        "s.close()"
+
+    cmd = "python -c \"{}\"".format(
+        script.format(
+            host_ip,
+            port,
+            iterations,
+            PAYLOAD_DATA_SIZE
+        )
+    )
+    netns_cmd = jailer.netns_cmd_prefix() + cmd
+    exit_code = subprocess.call(netns_cmd, shell=True)
+    assert exit_code == 0
+
+
+def _send_data_g2h(ssh_connection, host_ip, host_port, iterations, data,
+                   retries):
+    script =                                                        \
+        "import socket\n"                                           \
+        "import time\n"                                             \
+        "s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)\n"   \
+        "retries={}\n"                                              \
+        "while retries > 0:\n"                                      \
+        "   try:\n"                                                 \
+        "       s.connect(('{}',{}))\n"                             \
+        "       retries = 0\n"                                      \
+        "   except Exception as e:\n"                               \
+        "       retries -= 1\n"                                     \
+        "       time.sleep(1)\n"                                    \
+        "       if retries == 0:\n"                                 \
+        "           exit(1)\n"                                      \
+        "send_iterations={}\n"                                      \
+        "while send_iterations > 0:\n"                              \
+        "   s.sendall('{}')\n"                                      \
+        "   send_iterations -= 1\n"                                 \
+        "s.close()"
+
+    cmd = "python -c \"{}\"".format(
+        script.format(
+            retries,
+            host_ip,
+            str(host_port),
+            iterations,
+            data
+        )
+    )
+
+    # Wait server to initialize.
+    exit_code, _, stderr = ssh_connection.execute_command(cmd)
+    # If this assert fails, a connection refused happened.
+    assert exit_code == 0
+    assert stderr.read() == ''
+
+
+def _flush_metrics(test_microvm, metrics_fifo):
```

Comment:
nit: this could be moved to `host_tools` and used both here and in `test_flush_metrics`

Reply:
Done.